balancer/weightedtarget/weightedaggregator/aggregator.go

```diff
@@ -180,6 +191,9 @@ func (wbsa *Aggregator) UpdateState(id string, newState balancer.State) {
 		// it's either removed, or never existed.
 		return
 	}
+
+	wbsa.csEvltr.RecordTransition(oldState.stateToAggregate, newState.ConnectivityState)
```

Comment:
Shouldn't we be updating the connStateEvaluator after the `if` condition below? i.e, shouldn't we be filtering out a transition from `TF` to `Connecting`?

Reply:
Yes, good catch. Let's also make sure there's a test case covering this behavior since it's important and a little tricky. @arvindbr8